src/main/java/com/example/spring_bank_project/Account/Application/UseCase/AddNewUser/AddNewUserHandler.java

```diff
@@ -0,0 +1,29 @@
+package com.example.spring_bank_project.Account.Application.UseCase.AddNewUser;
+
+import com.example.spring_bank_project.Account.Domain.Entity.User;
+import com.example.spring_bank_project.Account.Domain.Factory.UserFactory;
+import com.example.spring_bank_project.Account.Domain.ValueObject.EncodedPin;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class AddNewUserHandler {
+    private final UserFactory factory;
+    private final NewUserPersistence persistence;
+    private final PinEncoder encoder;
+
+    public void handle(AddNewUserCommand command) {
+        AddNewUserContract contract = command.getContract();
+
+        EncodedPin encodedPin = this.encoder.encode(contract.getPin());
```

Comment:
this.poleKlasy mozna zamienic na po prostu poleKlasy ale ofc to kwestia preferencji

Reply:
wolę użyć thisa, dzięki temu mam wizualne rozróżnienie co jest lokalną zmienną a co property klasy